core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -249,6 +248,7 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
         /* register broker metrics */
         registerStats()
 
+        brokerState.newState(RunningAsBroker)
```

Comment:
Yes, that's a good point. It seems that it's more natural to start the controller after the broker registration in ZK. Could you file a separate jira to track that?